sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -132,9 +133,14 @@ class AccountState(val params: NetworkParams,
 
           val txHash = idToBytes(ethTx.id)
 
-          val contractAddress = if (ethTx.getTo != null && !stateView.isEoaAccount(ethTx.getTo.address()) ) {
-            ethTx.getTo.address()
+          // The contract address created, if the transaction was a contract creation
+          val contractAddress = if (ethTx.getTo == null) {
+            // this w3j util method is equivalent to the createAddress() in geth triggered also by CREATE opcode.
+            // Note: geth has also a CREATE2 opcode which may be optionally used in a smart contract solidity implementation
+            // to deploy another contract with a predefined address.
+            generateContractAddress(ethTx.getFrom.address, ethTx.getNonce)
           } else {
+            // otherwise a zero filled address
             new Array[Byte](Address.LENGTH)
```

Comment:
I think in case to is not null we should return empty array, not a 20 bytes of zeros. 20 zeros is a valid contract address name.
official docs say that we need to return `null` https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_gettransactionreceipt.
Check how `null` is encoded